### ISSUE TITLE
new lighting messes with thrall Darksight

### ISSUE THREAD

**Issue by Spacedong** [HippieStationCode/HippieStation13#1819]
http://vgy.me/K4wpmV.png

I don't know if this is normal or not, it looks really weird

EDIT: the same room without darksight http://vgy.me/h8GEWd.png

**Comment by Spacedong** [comment]
so is it a feature or a bug

**Comment by Kokojos** [comment]
new feature causes bug.

**Comment by blobbernaut** [comment]
I think they get full vision of it if there's any amount of light at all.

**Comment by JustinBNewton** [comment]
Can someone clarify how darksight is supposed to function exactly?

**Comment by Kokojos** [comment]
See everything in the dark. Right now you don't see shit but yourself and mobs, and where light is over 0, like near space.

This really needs a fix, shadowlings should opperate in the shadows.

**Comment by Grufus2** [comment]
Here to explain this bug for a coder to fix it.

In the file /code/game/gamemodes/shadowling/shadowling_abilities.dm
We can see this piece of code at line 721:
```
/obj/effect/proc_holder/spell/self/thrall_vision/cast(mob/living/carbon/human/user)
	active = !active
	if(active)
		user << "<span class='notice'>You shift the nerves in your eyes, allowing you to see in the dark.</span>"
		user.see_in_dark = 8
		user.dna.species.invis_sight = SEE_INVISIBLE_MINIMUM
	else
		user << "<span class='notice'>You return your vision to normal.</span>"
		user.see_in_dark = 3
		user.dna.species.invis_sight = initial(user.dna.species.invis_sight)
```

The culprit is the see_in_dark. If you activate the spell, you will see that the variable didn't change(or changed just for a very short time)

The reason this is happening is because of the handle_vision proc in the following location
/code/modules/mob/living/carbon/human/species.dm(Line 664)

```
/datum/species/proc/handle_vision(mob/living/carbon/human/H)
...
Line 682: H.see_in_dark = (H.sight == SEE_TURFS|SEE_MOBS|SEE_OBJS) ? 3 : darksight
...
```

The why shadowlings doesn't get affected by this is because in their species they have another way to handle their vision.

One way to fix it is to check if the mob is a thrall after that line of code, if it is set their see_in_dark variable to 8.

Hope this helps.

**Comment by danhibiki** [comment]
Thanks, that helps a lot. I knew it had something to do with the change I made in species, but I didn't know where to look for the shadowling change.

**Comment by Spacedong** [comment]
let it be known that I contributed to fixing shadowlings by reporting this

**Comment by Kokojos** [comment]
fixed by the late dan in #1896
